app/controllers/incomeSources/add/AddIncomeSourceStartDateController.scala

```diff
@@ -183,69 +148,106 @@ class AddIncomeSourceStartDateController @Inject()(authenticate: AuthenticationP
-          controllers.incomeSources.add.routes.AddIncomeSourceController.showAgent(),
-          controllers.incomeSources.add.routes.AddIncomeSourceStartDateController.submitUKPropertyAgent,
-          controllers.incomeSources.add.routes.AddIncomeSourceStartDateCheckController.showUKPropertyAgent
-        )
-      case (true, ForeignProperty) =>
-        (
-          controllers.incomeSources.add.routes.AddIncomeSourceController.showAgent(),
-          controllers.incomeSources.add.routes.AddIncomeSourceStartDateController.submitForeignPropertyAgent,
-          controllers.incomeSources.add.routes.AddIncomeSourceStartDateCheckController.showForeignPropertyAgent
+  private def getFilledForm(form: Form[DateFormElement],
+                            incomeSourceType: IncomeSourceType,
+                            isUpdate: Boolean)(implicit user: MtdItUser[_]): Either[Throwable, Form[DateFormElement]] = {
+
+    val maybeStartDate = getStartDate(incomeSourceType)
+
+    (maybeStartDate, isUpdate) match {
+      case (Some(date), true) if Try(LocalDate.parse(date)).toOption.isDefined =>
+        Right(
+          form.fill(
+            DateFormElement(
+              LocalDate.parse(date)
+            )
+          )
         )
+      case (Some(date), true) =>
+        Left(new Error(s"Could not parse $date as a LocalDate"))
+      case _ => Right(form)
     }
   }
 
-  def changeBusinessStartDate(): Action[AnyContent] = Action(Ok("Change Business Start Date WIP"))
-  def changeBusinessStartDateAgent(): Action[AnyContent] = Action(Ok("Agent Change Business Start Date WIP"))
-  def changeUKPropertyStartDate(): Action[AnyContent] = Action(Ok("Change UK Property Start Date WIP"))
-  def changeUKPropertyStartDateAgent(): Action[AnyContent] = Action(Ok("Agent Change UK Property Start Date WIP"))
-  def changeForeignPropertyStartDate(): Action[AnyContent] = Action(Ok("Change Foreign Property Start Date WIP"))
-  def changeForeignPropertyStartDateAgent(): Action[AnyContent] = Action(Ok("Agent Change Foreign Property Start Date WIP"))
+  private def getStartDate(incomeSourceType: IncomeSourceType)(implicit user: MtdItUser[_]): Option[String] = {
+    user.session.get(incomeSourceType.startDateSessionKey)
+  }
+
+  private def getCalls(incomeSourceType: IncomeSourceType,
+                       isAgent: Boolean,
+                       isChange: Boolean): (Call, Call, Call) = {
+
+    val postAction = routes.AddIncomeSourceStartDateController.submit(incomeSourceType.key, isAgent, isChange)
+
+    val successCall = routes.AddIncomeSourceStartDateCheckController.show(incomeSourceType.key, isAgent, isChange)
+
+    (
+      getBackUrl(isAgent, isChange, incomeSourceType),
+      postAction,
+      successCall
+    )
+  }
+
+  def getBackUrl(isAgent: Boolean, isChange: Boolean, incomeSourceType: IncomeSourceType): Call = {
+    if(isAgent)
+      agentCalls(isChange)(incomeSourceType)
+    else
+      individualCalls(isChange)(incomeSourceType)
+  }
+
```

Comment:
call by name would work better in this context, but thats fine, lets leave these changes as is